app/src/main/java/com/android/wildex/ui/social/FriendScreenViewModel.kt

```diff
@@ -0,0 +1,351 @@
+package com.android.wildex.ui.social
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.wildex.model.RepositoryProvider
+import com.android.wildex.model.relationship.Relationship
+import com.android.wildex.model.relationship.RelationshipRepository
+import com.android.wildex.model.relationship.StatusEnum
+import com.android.wildex.model.user.SimpleUser
+import com.android.wildex.model.user.UserFriendsRepository
+import com.android.wildex.model.user.UserRepository
+import com.android.wildex.model.utils.Id
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/** Represents the UI state of the Friend Screen */
+data class FriendsScreenUIState(
+    val friends: List<FriendState> = emptyList(),
+    val suggestions: List<RecommendationResult> = emptyList(),
+    val receivedRequests: List<Relationship> = emptyList(),
+    val sentRequests: List<Relationship> = emptyList(),
```

Comment:
Use the future new `FriendRequest` data class
```suggestion
    val receivedRequests: List<FriendRequest> = emptyList(),
    val sentRequests: List<FriendRequest> = emptyList(),
```